Ghidra/Framework/SoftwareModeling/src/main/java/ghidra/app/plugin/assembler/sleigh/sem/AssemblyResolvedBackfill.java

```diff
@@ -139,7 +139,19 @@ public AssemblyResolution solve(RecursiveDescentSolver solver, Map<String, Long>
 				return ar;
 			}
 			AssemblyResolvedPatterns rc = (AssemblyResolvedPatterns) ar;
-			return rc.shift(offset);
+			AssemblyResolvedPatterns shifted = rc.shift(offset);
+			
+			// get the operand data structure of the operand we want to populate info with
+			AssemblyOperandData od = cur.operandData.findDescription(description);
+			// populate operand data node with mask, value, shift, and expression for
+			// backfill operands this is similar code for operand data population for normal
+			// operands in AssemblyOperandState
+			od.setMask(rc.ins.getMask());
```

Comment:
I'm fairly sure we were running into problems without doing this work here in the backfill. I think because of the shift that the backfill is doing, if we don't also do that shift in our AssemblyOperandData our data is incorrect. I think this might also be the answer to why we can't always do everything on the first visit.

It's worth mentioning here that our understanding of the how (and why) of backfills (and context) is very limited, so we might be missing some much better more obvious ways of accomplishing what we're trying to do.